### PR TITLE
Minor change to insert, switch, replace, RingMap acting on lists

### PR DESCRIPTION
**Pull request by pzinn**
This PR fixes two inconsistencies in allowed types:
- one can now write `f \ l` for `f` a `RingMap` and `l` any `VisibleList`, not just `List` (same as `f \ l` when `f` is a `Function`)
- `insert`, `switch` and `replace`, which are all based on `take`, accept `BasicList`, not just `VisibleList`, like `take` itself.

### CODE REVIEW

**Comment by pzinn** [comment]
can this be reviewed/merged? I have other PRs coming that depend on this.

File: M2/Macaulay2/editors/emacs
Comment (d-torrance) [inline M2/Macaulay2/editors/emacs:1]: Did you mean to update the emacs submodule?

Comment (pzinn) [reply M2/Macaulay2/editors/emacs:1]: argh! no

**Comment by d-torrance** [comment]
I've always interpreted `BasicList` objects as things we don't really think of as lists, but just happen to be implemented as lists. In particular, I'm not sure how many "listy" methods outside of the ones defined in the interpreter we want installed for them.

**Comment by pzinn** [comment]
well, `replace` is pretty useful no matter what type of `BasicList` you're dealing with.

**Comment by d-torrance** [comment]
> well, `replace` is pretty useful no matter what type of `BasicList` you're dealing with.

Sounds good to me!

**Review by d-torrance — APPROVED** [review]

**Pull request merged**